src/coupling/generic_assembly.hh

```diff
@@ -136,83 +139,29 @@ public:
-            element_cache_map_(cell);
-
-            if ( cell.is_own() && (active_integrals_ & ActiveIntegrals::bulk) ) {
-                START_TIMER("assemble_volume_integrals");
-                for (i=0; i<integrals_size_[0]; ++i) { // volume integral
-                    switch (bulk_integral_data_[i].cell.dim()) {
-                    case 1:
-                        multidim_assembly_[1_d]->assemble_volume_integrals(bulk_integral_data_[i].cell);
-                        break;
-                    case 2:
-                        multidim_assembly_[2_d]->assemble_volume_integrals(bulk_integral_data_[i].cell);
-                        break;
-                    case 3:
-                        multidim_assembly_[3_d]->assemble_volume_integrals(bulk_integral_data_[i].cell);
-                        break;
-                    }
-                }
-                END_TIMER("assemble_volume_integrals");
-            }
-
-            if ( cell.is_own() && (active_integrals_ & ActiveIntegrals::boundary) ) {
-                START_TIMER("assemble_fluxes_boundary");
-                for (i=0; i<integrals_size_[3]; ++i) { // boundary integral
-                    switch (boundary_integral_data_[i].side.dim()) {
-                    case 1:
-                        multidim_assembly_[1_d]->assemble_fluxes_boundary(boundary_integral_data_[i].side);
-                        break;
-                    case 2:
-                        multidim_assembly_[2_d]->assemble_fluxes_boundary(boundary_integral_data_[i].side);
-                        break;
-                    case 3:
-                        multidim_assembly_[3_d]->assemble_fluxes_boundary(boundary_integral_data_[i].side);
-                        break;
-                    }
-                }
-                END_TIMER("assemble_fluxes_boundary");
+            if (!add_into_patch) {
+        	    element_cache_map_.start_elements_update();
+        	    add_into_patch = true;
             }
 
-            if (active_integrals_ & ActiveIntegrals::edge) {
-                START_TIMER("assemble_fluxes_elem_elem");
-                for (i=0; i<integrals_size_[1]; ++i) { // edge integral
-                    switch (edge_integral_data_[i].edge_side_range.begin()->dim()) {
-                    case 1:
-                        multidim_assembly_[1_d]->assemble_fluxes_element_element(edge_integral_data_[i].edge_side_range);
-                        break;
-                    case 2:
-                        multidim_assembly_[2_d]->assemble_fluxes_element_element(edge_integral_data_[i].edge_side_range);
-                        break;
-                    case 3:
-                        multidim_assembly_[3_d]->assemble_fluxes_element_element(edge_integral_data_[i].edge_side_range);
-                        break;
-                    }
-                }
-                END_TIMER("assemble_fluxes_elem_elem");
-            }
+            this->add_integrals_of_computing_step(cell);
```

Comment:
We have to check that we fit into the preallocated value cache size. 
1. move creation of the ElementCacheMap into Generic assembly
2. add triples (i_reg, i_element, i_point) into a list
3. similarly add to integral lists
4. for every list store: `temporary_size` and `final_size`
5. if the new point or new ghost patch element does not fit into preallocated arrays we reset finish the patch using the final_size, otherwise we set final_size to the temporary_size after processing a single cell.

Reply:
Try to make auxiliary template class for a list with temporary size.